mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,268 @@
+update_url: 'projects/{{project}}/locations/{{region}}/indexEndpoints/{{index_endpoint}}:mutateDeployedIndex'
+delete_url: 'projects/{{project}}/locations/{{region}}/indexEndpoints/{{index_endpoint}}:undeployIndex'
+self_link: 'projects/{{project}}/locations/{{region}}/indexEndpoints/{{index_endpoint}}'
+import_format: ['projects/{{project}}/locations/{{region}}/indexEndpoints/{{index_endpoint}}/deployedIndex/{{deployed_index_id}}']
+id_format: 'projects/{{project}}/locations/{{region}}/indexEndpoints/{{index_endpoint}}/deployedIndex/{{deployed_index_id}}'
+update_verb: :POST
+delete_verb: :POST
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.indexEndpoints#DeployedIndex
+async: !ruby/object:Api::OpAsync
+  actions:
+    - create
+    - delete
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 45
+      update_minutes: 45
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+description: |-
+  An endpoint indexes are deployed into. An index endpoint can have multiple deployed indexes.
+autogen_async: false
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vertex_ai_index_endpoint_deployed_index_basic"
+    primary_resource_id: "basic_deployed_index"
+    vars:
+      project: 'vertex-ai'
+      endpoint_name: 'endpoint-name'
+      network_name: 'network-name'
+      display_name: "vertex-deployed-index"
+      display_name_index: 'test-index'
+      bucket_name: "bucket-name"
+      address_name: 'vertex-ai-range'
+    test_vars_overrides:
+      network_name: 'acctest.BootstrapSharedServiceNetworkingConnection(t, "vpc-network-1")'
+      address_name: 'acctest.BootstrapSharedTestGlobalAddress(t, "vpc-network-1", acctest.AddressWithPrefixLength(8))'
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vertex_ai_index_endpoint_deployed_index_dedicated_resources"
+    primary_resource_id: "dedicated_resources"
+    vars:
+      display_name_index: 'test-index'
+      bucket_name: "bucket-name"
+      display_name: "vertex-deployed-index"
+    skip_docs: true
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/vertex_ai_index_endpoint_deployed_index.go.erb
+  update_encoder: templates/terraform/update_encoder/vertex_ai_index_endpoint_deployed_index.go.erb
+  decoder: templates/terraform/decoders/vertex_ai_index_endpoint_deployed_index.go.erb
+  pre_delete: templates/terraform/pre_delete/vertex_ai_index_endpoint_deployed_index.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: region
```

Comment:
[This is the related comment](https://github.com/GoogleCloudPlatform/magic-modules/pull/11039#discussion_r1666841675) - if we decide the schema should contain region then it should be updated to include project too, to avoid the bug I described